pkg/apiserver/cubeapi/user/auth.go

```diff
@@ -156,3 +119,123 @@ func Login(c *gin.Context) {
 	response.SuccessReturn(c, user)
 	return
 }
+
+func GitHubLogin(c *gin.Context) {
+	code := c.Query("code")
+	if code == "" {
+		clog.Error("code is null")
+		response.FailReturn(c, errcode.AuthenticateError)
+		return
+	}
+
+	g := github.GetProvider()
+	if !g.GitHubIsEnable {
+		clog.Error("github auth is disabled")
+		response.FailReturn(c, errcode.AuthenticateError)
+		return
+	}
+
+	userInfo, err := g.IdentityExchange(code)
+	if err != nil {
+		response.FailReturn(c, errcode.AuthenticateError)
+		return
+	}
+	clog.Info("user %s auth success by github", userInfo.GetUserName())
+
+	// get user by name
+	userName := hex.EncodeToString([]byte(userInfo.GetUserName()))
```

Comment:
Why we need hex?

Reply:
github name may not be allowed to store to K8s，such as capital letters